WordPress/gutenberg
# [RNMobile] Add navigation to link settings and add link picker screen to block settings

GB mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2742

In this PR i added a link picker screen to the link settings. It is used in Button and Social Icons.
I also added the navigation inside the BottomSheet for LinkSettings with `withBottomSheet` set to true.

There is also a prop  in LinkSettingsNavigaiton `hasPicker` to determine if we would like to use link picker or regular text input

## How has this been tested?
Case 1:
- select Button block
- open block settings and click on the link URL
- the new screen should be pushed
- test it the same as regular link picker

Case 2:
- select Social Icon block
- open link settings
- the default text input should be used instead of pushing a new screen

## Screenshots 
![link-picker](https://user-images.githubusercontent.com/16336501/96271707-7b85a300-0fcd-11eb-91f1-4423c52cafa5.gif)

## Types of changes
New Feature

## Checklist:
